Makefile

```diff
@@ -107,6 +107,9 @@ export
 rl_clientlib: vw
 	cd reinforcement_learning/rlclientlib; $(MAKE) -j $(NPROCS) things
 
+# O3 turns on the devirtualize optimization option. This is not compatible with mock testing as some
+# virtual methods can be devirtualized if possible.
+rl_clientlib_test: FLAGS:=$(filter-out -O3,$(FLAGS))
```

Comment:
Is there a better solution that works with O3?

Reply:
Yes! We can just use -fno-devirtualize to scope the change to the single optimization that breaks it